### ISSUE TITLE
hakyll-4.7.5.2 compile error with ghc 7.8.x

### ISSUE THREAD

**Issue by peti** [jaspervdj/hakyll#433]
Citing from https://travis-ci.org/peti/quizmaster/jobs/135657003:

~~~~
[42 of 47] Compiling Hakyll.Web.Tags  ( src/Hakyll/Web/Tags.hs, dist/build/Hakyll/Web/Tags.o )

src/Hakyll/Web/Tags.hs:101:17:
    No instance for (Show (String -> Identifier))
      arising from the second field of â€˜Tagsâ€™ (type â€˜String
                                                     -> Identifierâ€™)
    Possible fix:
      use a standalone 'deriving instance' declaration,
        so you can specify the instance context yourself
    When deriving the instance for (Show Tags)
~~~~

**Comment by jaspervdj-luminal** [comment]
This was fixed in `4.8.3.0`. Is it possible to upgrade?

**Comment by peti** [comment]
Hmm, I didn't restrict the version of `hakyll` in my Cabal file, but apparently the resolver hasn't picked up the new version for some reason?

**Comment by jonkri** [comment]
I also got this now after a Cabal update in a fresh sandbox using GHC 7.8.

Here you can see the exact versions being pulled:

    [jonkri@localhost ~]$ mkdir hakyll
    [jonkri@localhost ~]$ cd hakyll/
    [jonkri@localhost hakyll]$ cabal sandbox init
    Writing a default package environment file to
    /home/jonkri/hakyll/cabal.sandbox.config
    Creating a new sandbox at /home/jonkri/hakyll/.cabal-sandbox
    [jonkri@localhost hakyll]$ cabal update
    Downloading the latest package list from hackage.haskell.org
    Note: there is a new version of cabal-install available.
    To upgrade, run: cabal install cabal-install
    [jonkri@localhost hakyll]$ cabal install hakyll
    Resolving dependencies...
    Notice: installing into a sandbox located at
    /home/jonkri/hakyll/.cabal-sandbox
    Configuring HUnit-1.3.1.1...
    Downloading SHA-1.6.4.2...
    Configuring base64-bytestring-1.0.0.1...
    Downloading byteable-0.1.1...
    Configuring cmdargs-0.10.14...
    Configuring digest-0.0.1.2...
    Downloading bytestring-builder-0.10.8.1.0...
    Building HUnit-1.3.1.1...
    Building cmdargs-0.10.14...
    Building base64-bytestring-1.0.0.1...
    Building digest-0.0.1.2...
    Downloading bytestring-mmap-0.2.2...
    Downloading data-default-class-0.1.0...
    Downloading deepseq-generics-0.1.1.2...
    Downloading extensible-exceptions-0.1.1.4...
    Downloading haddock-library-1.4.2...
    Downloading hinotify-0.3.8.1...
    Downloading hourglass-0.2.10...
    Installed digest-0.0.1.2
    Downloading hslua-0.4.1...
    Configuring dlist-0.7.1.2...
    Downloading memory-0.13...
    Downloading monads-tf-0.1.0.3...
    Configuring fail-4.9.0.0...
    Installed base64-bytestring-1.0.0.1
    Downloading mtl-2.1.3.1...
    Building dlist-0.7.1.2...
    Downloading rfc5051-0.1.0.3...
    Downloading setenv-0.1.1.3...
    Downloading split-0.2.3.1...
    Downloading time-locale-compat-0.1.1.2...
    Building fail-4.9.0.0...
    Downloading unix-compat-0.4.2.0...
    Configuring network-2.6.2.1...
    Installed HUnit-1.3.1.1
    Configuring primitive-0.6.1.0...
    Installed fail-4.9.0.0
    Downloading cereal-0.5.2.0...
    Configuring random-1.1...
    Installed dlist-0.7.1.2
    Building primitive-0.6.1.0...
    Building random-1.1...
    Building network-2.6.2.1...
    Configuring stm-2.4.4.1...
    Installed primitive-0.6.1.0
    Building stm-2.4.4.1...
    Configuring syb-0.6...
    Installed random-1.1
    Configuring tagged-0.8.4...
    Installed stm-2.4.4.1
    Building syb-0.6...
    Building tagged-0.8.4...
    Configuring text-1.2.2.1...
    Installed cmdargs-0.10.14
    Building text-1.2.2.1...
    Configuring utf8-string-1.0.1.1...
    Installed syb-0.6
    Downloading hs-bibutils-5.5...
    Building utf8-string-1.0.1.1...
    Configuring zlib-0.6.1.1...
    Installed tagged-0.8.4
    Building zlib-0.6.1.1...
    Configuring byteable-0.1.1...
    Installed network-2.6.2.1
    Building byteable-0.1.1...
    Configuring SHA-1.6.4.2...
    Installed byteable-0.1.1
    Building SHA-1.6.4.2...
    Configuring bytestring-mmap-0.2.2...
    Installed utf8-string-1.0.1.1
    Building bytestring-mmap-0.2.2...
    Configuring data-default-class-0.1.0...
    Installed bytestring-mmap-0.2.2
    Building data-default-class-0.1.0...
    Configuring deepseq-generics-0.1.1.2...
    Installed zlib-0.6.1.1
    Downloading zlib-bindings-0.1.1.5...
    Building deepseq-generics-0.1.1.2...
    Configuring extensible-exceptions-0.1.1.4...
    Installed data-default-class-0.1.0
    Building extensible-exceptions-0.1.1.4...
    Configuring haddock-library-1.4.2...
    Installed deepseq-generics-0.1.1.2
    Building haddock-library-1.4.2...
    Configuring hinotify-0.3.8.1...
    Installed extensible-exceptions-0.1.1.4
    Building hinotify-0.3.8.1...
    Installed hinotify-0.3.8.1
    Configuring hourglass-0.2.10...
    Building hourglass-0.2.10...
    Installed haddock-library-1.4.2
    Configuring hslua-0.4.1...
    Building hslua-0.4.1...
    Configuring memory-0.13...
    Installed hourglass-0.2.10
    Building memory-0.13...
    Configuring monads-tf-0.1.0.3...
    Installed memory-0.13
    Downloading asn1-types-0.3.2...
    Downloading cryptonite-0.16...
    Building monads-tf-0.1.0.3...
    Configuring bytestring-builder-0.10.8.1.0...
    Installed monads-tf-0.1.0.3
    Downloading MonadCatchIO-transformers-0.3.1.3...
    Building bytestring-builder-0.10.8.1.0...
    Configuring mtl-2.1.3.1...
    Installed bytestring-builder-0.10.8.1.0
    Building mtl-2.1.3.1...
    Configuring setenv-0.1.1.3...
    Installed mtl-2.1.3.1
    Downloading pem-0.2.2...
    Downloading regex-base-0.93.2...
    Building setenv-0.1.1.3...
    Configuring rfc5051-0.1.0.3...
    Installed hslua-0.4.1
    Building rfc5051-0.1.0.3...
    Configuring split-0.2.3.1...
    Installed setenv-0.1.1.3
    Building split-0.2.3.1...
    Configuring time-locale-compat-0.1.1.2...
    Installed split-0.2.3.1
    Building time-locale-compat-0.1.1.2...
    Configuring unix-compat-0.4.2.0...
    Installed rfc5051-0.1.0.3
    Configuring cereal-0.5.2.0...
    Installed time-locale-compat-0.1.1.2
    Building unix-compat-0.4.2.0...
    Building cereal-0.5.2.0...
    Configuring vector-0.11.0.0...
    Installed unix-compat-0.4.2.0
    Downloading filemanip-0.3.6.3...
    Building vector-0.11.0.0...
    Configuring StateVar-1.1.0.4...
    Installed text-1.2.2.1
    Downloading cmark-0.5.2.1...
    Downloading enumerator-0.4.20...
    Downloading parsec-3.1.11...
    Downloading system-filepath-0.4.13.4...
    Downloading tagsoup-0.13.10...
    Downloading xml-1.3.14...
    Downloading zip-archive-0.3.0.2...
    Building StateVar-1.1.0.4...
    Configuring async-2.1.0...
    Installed cereal-0.5.2.0
    Downloading socks-0.5.5...
    Building async-2.1.0...
    Configuring hs-bibutils-5.5...
    Installed StateVar-1.1.0.4
    Building hs-bibutils-5.5...
    Configuring zlib-bindings-0.1.1.5...
    Installed async-2.1.0
    Downloading fsnotify-0.2.1...
    Building zlib-bindings-0.1.1.5...
    Configuring data-default-instances-base-0.1.0...
    Installed zlib-bindings-0.1.1.5
    Building data-default-instances-base-0.1.0...
    Configuring data-default-instances-containers-0.0.1...
    Installed data-default-instances-base-0.1.0
    Building data-default-instances-containers-0.0.1...
    Configuring data-default-instances-dlist-0.0.1...
    Installed data-default-instances-containers-0.0.1
    Building data-default-instances-dlist-0.0.1...
    Configuring data-default-instances-old-locale-0.0.1...
    Installed data-default-instances-dlist-0.0.1
    Building data-default-instances-old-locale-0.0.1...
    Installed data-default-instances-old-locale-0.0.1
    Downloading data-default-0.5.3...
    Configuring asn1-types-0.3.2...
    Building asn1-types-0.3.2...
    Configuring cryptonite-0.16...
    Installed asn1-types-0.3.2
    Downloading asn1-encoding-0.9.3...
    Building cryptonite-0.16...
    Configuring MonadCatchIO-transformers-0.3.1.3...
    Installed hs-bibutils-5.5
    Building MonadCatchIO-transformers-0.3.1.3...
    Configuring transformers-compat-0.5.1.4...
    Installed MonadCatchIO-transformers-0.3.1.3
    Building transformers-compat-0.5.1.4...
    Configuring pem-0.2.2...
    Installed transformers-compat-0.5.1.4
    Building pem-0.2.2...
    Configuring regex-base-0.93.2...
    Installed pem-0.2.2
    Building regex-base-0.93.2...
    Configuring filemanip-0.3.6.3...
    Installed regex-base-0.93.2
    Downloading regex-pcre-builtin-0.94.4.8.8.35...
    Downloading regex-posix-0.95.2...
    Building filemanip-0.3.6.3...
    Configuring blaze-builder-0.4.0.2...
    Installed filemanip-0.3.6.3
    Building blaze-builder-0.4.0.2...
    Configuring hashable-1.2.4.0...
    Installed SHA-1.6.4.2
    Building hashable-1.2.4.0...
    Configuring mime-types-0.1.0.7...
    Installed blaze-builder-0.4.0.2
    Downloading streaming-commons-0.1.15.5...
    Building mime-types-0.1.0.7...
    Configuring xml-types-0.3.6...
    Installed hashable-1.2.4.0
    Downloading unordered-containers-0.2.7.1...
    Building xml-types-0.3.6...
    Configuring enumerator-0.4.20...
    Installed cryptonite-0.16
    Downloading cryptohash-0.11.9...
    Configuring cmark-0.5.2.1...
    Installed xml-types-0.3.6
    Building enumerator-0.4.20...
    Configuring parsec-3.1.11...
    Installed vector-0.11.0.0
    Downloading JuicyPixels-3.2.7.1...
    Downloading scientific-0.3.4.7...
    Building cmark-0.5.2.1...
    Building parsec-3.1.11...
    Configuring system-filepath-0.4.13.4...
    Installed mime-types-0.1.0.7
    Building system-filepath-0.4.13.4...
    Configuring tagsoup-0.13.10...
    Installed system-filepath-0.4.13.4
    Building tagsoup-0.13.10...
    Configuring zip-archive-0.3.0.2...
    Installed parsec-3.1.11
    Downloading regex-tdfa-1.2.2...
    Configuring xml-1.3.14...
    Installed enumerator-0.4.20
    Downloading zlib-enum-0.2.3.1...
    Building xml-1.3.14...
    Building zip-archive-0.3.0.2...
    Configuring socks-0.5.5...
    Installed xml-1.3.14
    Building socks-0.5.5...
    Configuring fsnotify-0.2.1...
    Installed zip-archive-0.3.0.2
    Installed cmark-0.5.2.1
    Building fsnotify-0.2.1...
    Configuring data-default-0.5.3...
    Building data-default-0.5.3...
    Configuring asn1-encoding-0.9.3...
    Installed tagsoup-0.13.10
    Configuring exceptions-0.8.2.1...
    Installed data-default-0.5.3
    Building asn1-encoding-0.9.3...
    Configuring mmorph-1.0.6...
    Installed fsnotify-0.2.1
    Building exceptions-0.8.2.1...
    Configuring transformers-base-0.4.4...
    Building mmorph-1.0.6...
    Installed socks-0.5.5
    Building transformers-base-0.4.4...
    Configuring regex-posix-0.95.2...
    Installed mmorph-1.0.6
    Configuring regex-pcre-builtin-0.94.4.8.8.35...
    Installed exceptions-0.8.2.1
    Building regex-posix-0.95.2...
    Configuring blaze-markup-0.7.0.3...
    Installed transformers-base-0.4.4
    Building regex-pcre-builtin-0.94.4.8.8.35...
    Configuring cookie-0.4.2...
    Installed asn1-encoding-0.9.3
    Downloading asn1-parse-0.9.4...
    Building blaze-markup-0.7.0.3...
    Building cookie-0.4.2...
    Configuring streaming-commons-0.1.15.5...
    Installed cookie-0.4.2
    Configuring nats-1.1...
    Installed regex-posix-0.95.2
    Building streaming-commons-0.1.15.5...
    Building nats-1.1...
    Configuring unordered-containers-0.2.7.1...
    Installed blaze-markup-0.7.0.3
    Configuring cryptohash-0.11.9...
    Installed nats-1.1
    Building unordered-containers-0.2.7.1...
    Building cryptohash-0.11.9...
    Configuring JuicyPixels-3.2.7.1...
    Installed streaming-commons-0.1.15.5
    Downloading blaze-builder-enumerator-0.2.1.0...
    Building JuicyPixels-3.2.7.1...
    Configuring scientific-0.3.4.7...
    Installed cryptohash-0.11.9
    Building scientific-0.3.4.7...
    Configuring network-uri-2.6.1.0...
    Installed unordered-containers-0.2.7.1
    Building network-uri-2.6.1.0...
    Configuring regex-tdfa-1.2.2...
    Installed scientific-0.3.4.7
    Building regex-tdfa-1.2.2...
    Configuring zlib-enum-0.2.3.1...
    Installed network-uri-2.6.1.0
    Downloading HTTP-4000.3.3...
    Installed regex-pcre-builtin-0.94.4.8.8.35
    Configuring temporary-1.2.0.4...
    Building zlib-enum-0.2.3.1...
    Building temporary-1.2.0.4...
    Configuring monad-control-1.0.1.0...
    Installed zlib-enum-0.2.3.1
    Building monad-control-1.0.1.0...
    Configuring asn1-parse-0.9.4...
    Installed temporary-1.2.0.4
    Building asn1-parse-0.9.4...
    Configuring blaze-html-0.8.1.1...
    Installed monad-control-1.0.1.0
    Configuring blaze-builder-enumerator-0.2.1.0...
    Installed asn1-parse-0.9.4
    Downloading x509-1.6.3...
    Building blaze-html-0.8.1.1...
    Building blaze-builder-enumerator-0.2.1.0...
    Configuring semigroups-0.18.1...
    Installed blaze-builder-enumerator-0.2.1.0
    Building semigroups-0.18.1...
    Configuring attoparsec-0.13.0.2...
    Installed semigroups-0.18.1
    Building attoparsec-0.13.0.2...
    Configuring HTTP-4000.3.3...
    Installed blaze-html-0.8.1.1
    Downloading highlighting-kate-0.6.2...
    Building HTTP-4000.3.3...
    Installed HTTP-4000.3.3
    Configuring lifted-base-0.2.3.6...
    Building lifted-base-0.2.3.6...
    Configuring x509-1.6.3...
    Installed lifted-base-0.2.3.6
    Downloading enclosed-exceptions-1.0.1.1...
    Downloading resourcet-1.1.7.4...
    Building x509-1.6.3...
    Configuring case-insensitive-1.2.0.6...
    Installed x509-1.6.3
    Downloading x509-store-1.6.1...
    Building case-insensitive-1.2.0.6...
    Configuring void-0.7.1...
    Installed case-insensitive-1.2.0.6
    Downloading http-types-0.9.1...
    Building void-0.7.1...
    Configuring highlighting-kate-0.6.2...
    Installed attoparsec-0.13.0.2
    Downloading aeson-0.11.2.0...
    Downloading attoparsec-enumerator-0.3.4...
    Building highlighting-kate-0.6.2...
    Installed void-0.7.1
    Configuring enclosed-exceptions-1.0.1.1...
    Building enclosed-exceptions-1.0.1.1...
    Configuring resourcet-1.1.7.4...
    Installed enclosed-exceptions-1.0.1.1
    Building resourcet-1.1.7.4...
    Configuring x509-store-1.6.1...
    Installed resourcet-1.1.7.4
    Downloading conduit-1.2.6.6...
    Building x509-store-1.6.1...
    Configuring http-types-0.9.1...
    Installed x509-store-1.6.1
    Downloading x509-system-1.6.3...
    Downloading x509-validation-1.6.3...
    Building http-types-0.9.1...
    Configuring attoparsec-enumerator-0.3.4...
    Installed http-types-0.9.1
    Building attoparsec-enumerator-0.3.4...
    Configuring aeson-0.11.2.0...
    Installed JuicyPixels-3.2.7.1
    Building aeson-0.11.2.0...
    Configuring contravariant-1.4...
    Installed attoparsec-enumerator-0.3.4
    Downloading snap-core-0.9.8.0...
    Building contravariant-1.4...
    Installed contravariant-1.4
    Downloading lrucache-1.2.0.0...
    Configuring conduit-1.2.6.6...
    Building conduit-1.2.6.6...
    Configuring x509-system-1.6.3...
    Installed regex-tdfa-1.2.2
    Building x509-system-1.6.3...
    Configuring x509-validation-1.6.3...
    Installed x509-system-1.6.3
    Building x509-validation-1.6.3...
    Configuring http-client-0.4.28...
    Installed conduit-1.2.6.6
    Configuring snap-core-0.9.8.0...
    Installed x509-validation-1.6.3
    Downloading tls-1.3.8...
    Building http-client-0.4.28...
    Building snap-core-0.9.8.0...
    Installed http-client-0.4.28
    Configuring lrucache-1.2.0.0...
    Building lrucache-1.2.0.0...
    Configuring conduit-extra-1.1.13.1...
    Installed lrucache-1.2.0.0
    Building conduit-extra-1.1.13.1...
    Configuring tls-1.3.8...
    Installed aeson-0.11.2.0
    Downloading pandoc-types-1.16.1...
    Downloading yaml-0.8.17.2...
    Building tls-1.3.8...
    Installed conduit-extra-1.1.13.1
    Downloading xml-conduit-1.3.5...
    Configuring aeson-pretty-0.7.2...
    Building aeson-pretty-0.7.2...
    Configuring pandoc-types-1.16.1...
    Installed aeson-pretty-0.7.2
    Building pandoc-types-1.16.1...
    Configuring yaml-0.8.17.2...
    Installed snap-core-0.9.8.0
    Downloading snap-server-0.9.5.1...
    Building yaml-0.8.17.2...
    Installed tls-1.3.8
    Configuring xml-conduit-1.3.5...
    Downloading connection-0.2.5...
    Building xml-conduit-1.3.5...
    Configuring snap-server-0.9.5.1...
    Installed highlighting-kate-0.6.2
    Building snap-server-0.9.5.1...
    Configuring connection-0.2.5...
    Installed yaml-0.8.17.2
    Building connection-0.2.5...
    Installed pandoc-types-1.16.1
    Downloading texmath-0.8.6.3...
    Installed connection-0.2.5
    Downloading http-client-tls-0.2.4...
    Configuring http-client-tls-0.2.4...
    Configuring texmath-0.8.6.3...
    Building http-client-tls-0.2.4...
    Building texmath-0.8.6.3...
    Installed xml-conduit-1.3.5
    Installed http-client-tls-0.2.4
    Downloading http-conduit-2.1.10.1...
    Configuring http-conduit-2.1.10.1...
    Building http-conduit-2.1.10.1...
    Installed http-conduit-2.1.10.1
    Installed snap-server-0.9.5.1
    Installed texmath-0.8.6.3
    Downloading pandoc-1.17.1...
    Configuring pandoc-1.17.1...
    Building pandoc-1.17.1...
    Installed pandoc-1.17.1
    Downloading pandoc-citeproc-0.9.1.1...
    Configuring pandoc-citeproc-0.9.1.1...
    Building pandoc-citeproc-0.9.1.1...
    Installed pandoc-citeproc-0.9.1.1
    Downloading hakyll-4.7.5.2...
    Configuring hakyll-4.7.5.2...
    Building hakyll-4.7.5.2...
    Failed to install hakyll-4.7.5.2
    Last 10 lines of the build log ( /home/jonkri/hakyll/.cabal-sandbox/logs/hakyll-4.7.5.2.log ):
    [42 of 47] Compiling Hakyll.Web.Tags  ( src/Hakyll/Web/Tags.hs, dist/dist-sandbox-50ff95f2/build/Hakyll/Web/Tags.o )
    
    src/Hakyll/Web/Tags.hs:101:17:
        No instance for (Show (String -> Identifier))
          arising from the second field of ‘Tags’ (type ‘String
                                                         -> Identifier’)
        Possible fix:
          use a standalone 'deriving instance' declaration,
            so you can specify the instance context yourself
        When deriving the instance for (Show Tags)
    cabal: Error: some packages failed to install:
    hakyll-4.7.5.2 failed during the building phase. The exception was:
    ExitFailure 1

**Comment by peti** [comment]
There is clearly some problem in the version constraints specified by the latest `hakyll` version, because GHC 8.0.1 cannot build it either: https://travis-ci.org/peti/quizmaster/jobs/135791408. Instead, it falls back to trying to compile hakyll-2.4.4.

**Comment by jaspervdj** [comment]
This should be fixed with the latest Hakyll, `4.8.3.2`.

**Comment by peti** [comment]
Yes, I can confirm that the new version builds correctly with older compilers. Thank you very much!
